Plugin.fs

```diff
@@ -38,15 +38,21 @@ type Plugin
         // there's no nice way to handle all the unmarked nulls in the API; options are:
         // - risk NullReferenceException and hope for the best
         // - "if not (isNull x) then" checks (don't use "<> null")
-        // - match and bind on the not-null case, as in this example (most idiomatic option)
-        match clientState.LocalPlayer with
-        | null -> ()
-        | localPlayer ->
-            match localPlayer.CurrentWorld.GameData with
-            | null -> ()
-            | world ->
-                chat.Print($"Hello {world.Name}!")
-                PluginLog.Log("Message sent successfully.")
+        // - match and bind on the not-null case
+        // - use Option.ofObj and handle it like a regular Option value
+        // - use the provided maybe monad and read all intermediate values using let!
+        //   the CE will short-circuit away should any let! find a null or None value
+        //   don't forget to pipe it to "ignore" if used imperatively as commands must return unit
+        maybe {
+            // the block only continues if the value of a let!, match!, do! or use! is not null
+            // should a result be null, the entire block automatically resolves to None
+            let! localPlayer = clientState.LocalPlayer
+            let! world = localPlayer.CurrentWorld.GameData
+
+            chat.Print $"Hello %s{world.Name.ToString()}!"
+            PluginLog.Log "Message sent successfully."
+        }
+        |> ignore
```

Comment:
I'd prefer if the compiler could help check that the CE produced `unit`, but I couldn't find any way to do it without forcing users to write an explicit `return! ()` inside the block which is still just `ignore` but now you have a non-monadic CE.